.tekton/tasks/update-dependencies.yaml

```diff
@@ -36,17 +36,25 @@ spec:
         && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
         && apt update \
         && apt install gh -y
-
+        apt install jq -y
+        
         gh auth login --with-token
-
+        
+        OPEN_PRS=$(gh pr list --head currency-bot --base main --state open --json number | jq 'length')
+  
         git ls-remote --exit-code --heads origin currency-bot >/dev/null 2>&1
-        EXIT_CODE=$?
+        BRANCH_EXISTS=$?
+
+        if [[ $BRANCH_EXISTS -eq 0 && $OPEN_PRS -eq 0]]; then
```

Comment:
suggestion: The renaming from `EXIT_CODE` to `BRANCH_EXISTS` is semantically wrong.
`BRANCH_EXISTS` of the value 0 means it exists. But it should be 1 to be readable. The usage of `EXIT_CODE` was more readable because its a known concept that 0 is positive.

```sh
if git ls-remote --exit-code --heads origin currency-bot >/dev/null 2>&1; then
    BRANCH_EXISTS=true
else
    BRANCH_EXISTS=false
fi

if [[ "$BRANCH_EXISTS" = true && $OPEN_PRS -eq 0 ]]; then
...
```